File: app/src/main/java/com/ddu/util/PopupUtils.java
package com.ddu.util;

import android.app.Activity;
import android.graphics.Rect;
import android.graphics.drawable.BitmapDrawable;
import android.support.annotation.NonNull;
import android.view.Gravity;
import android.view.View;
import android.widget.LinearLayout;
import android.widget.PopupWindow;

import com.ddu.R;


/**
 * Created by yzbzz on 16/4/18.
 */
public class PopupUtils {

    public static void showDefaultPopupWindow(@NonNull Activity activity, @NonNull View view, View popupView) {
        Rect frame = new Rect();
        activity.getWindow().getDecorView().getWindowVisibleDisplayFrame(frame);

        int[] location = new int[2];
        view.getLocationOnScreen(location);

        PopupWindow popupWindow = showPopupWindow(popupView);
        popupWindow.showAtLocation(view, Gravity.TOP | Gravity.RIGHT, 0, location[1] + frame.top - popupWindow.getHeight());
        popupWindow.update();

    }

    @NonNull
    public static PopupWindow showPopupWindow(View popupView) {
        PopupWindow popupWindow = new PopupWindow(popupView, LinearLayout.LayoutParams.WRAP_CONTENT, LinearLayout.LayoutParams.WRAP_CONTENT);
        popupWindow.setFocusable(true);
        popupWindow.setOutsideTouchable(true);
        popupWindow.setBackgroundDrawable(new BitmapDrawable());
        popupWindow.setAnimationStyle(R.style.MorePopupAnim);
        return popupWindow;
    }
}

File: icore/src/main/java/com/ddu/icore/util/PopupUtils.kt
package com.ddu.icore.util

import android.app.Activity
import android.graphics.Rect
import android.graphics.drawable.BitmapDrawable
import android.view.Gravity
import android.view.View
import android.widget.LinearLayout
import android.widget.PopupWindow

import com.ddu.icore.R


/**
 * Created by yzbzz on 16/4/18.
 */
object PopupUtils {

    fun showTopRightPopupWindow(activity: Activity, view: View, popupView: View) {
        val frame = Rect()
        activity.window.decorView.getWindowVisibleDisplayFrame(frame)

        val location = IntArray(2)
        view.getLocationOnScreen(location)

        val popupWindow = showPopupWindow(popupView)
        popupWindow.showAtLocation(view, Gravity.TOP or Gravity.RIGHT, 0, location[1] + frame.top - popupWindow.height)
        popupWindow.update()

    }

    fun showPopupWindow(popupView: View): PopupWindow {
        val popupWindow = PopupWindow(popupView, LinearLayout.LayoutParams.WRAP_CONTENT, LinearLayout.LayoutParams.WRAP_CONTENT)
        popupWindow.isFocusable = true
        popupWindow.isOutsideTouchable = true
        popupWindow.setBackgroundDrawable(BitmapDrawable())
        popupWindow.animationStyle = R.style.MorePopupAnim
        return popupWindow
    }
}
